### PR TITLE
Add support for trusted_domains to forward proxy mode

### PR DESCRIPTION
**Pull request by Jorilx**
## Summary

The trusted_domains settings is checked when in fetch proxy mode, but not while in forward proxy mode, see https://github.com/luckyPipewrench/pipelock/issues/276#issuecomment-4121351417

## Changes

Created a IsTrustedDomain() function, called inside ssrfSafeDialContext()

## Testing

- [x] `make test` passes
- [x] `make vet` passes
- [x] New tests added for new functionality
- [x] Tested manually (see comment on issue 276)

## Security Considerations

It can lower the scanning security, but it's entirely opt-in and configurable

## Summary by CodeRabbit

* **Bug Fixes**
  * SSRF protection now honors trusted-domain allowlists for hostname destinations while still blocking raw IP literals.

* **Tests**
  * Added tests verifying trusted-domain bypass behavior for hostnames.
  * Added tests for trusted-domain matching (wildcards, case/format normalization, and edge cases).

### CODE REVIEW

File: internal/scanner/scanner.go [hunk 1]
```diff
@@ -352,6 +352,18 @@ func (s *Scanner) IsInternalIP(ip net.IP) bool {
 	return false
 }
 
+// IsTrustedDomain checks if a hostname matches any trusted domain pattern.
+// Trusted domains bypass the SSRF internal-IP check.
+func (s *Scanner) IsTrustedDomain(hostname string) bool {
+	hostname = strings.ToLower(strings.TrimSuffix(hostname, "."))
```
Comment (DeathCamel58) [inline internal/scanner/scanner.go]: The `MatchDomain` function's first argument is already passed through `strings.ToLower`. There's no need to `ToLower` here.

File: internal/proxy/proxy.go [hunk 2]
```diff
@@ -642,7 +645,9 @@ func (p *Proxy) ssrfSafeDialContext(ctx context.Context, network, addr string) (
 			ip = v4
 		}
 		if currentSc.IsInternalIP(ip) {
-			return nil, fmt.Errorf("SSRF blocked: %s resolves to internal IP %s", host, ipStr)
+			if !currentSc.IsTrustedDomain(host) {
```
Comment (DeathCamel58) [inline internal/proxy/proxy.go]: Minor nitpick: IsTrustedDomain(host) is inside the IP loop, but the host doesn't change inside of the loop. Moving the check before the loop could avoid re-checking every resolved IP.

**Review by DeathCamel58 — APPROVED** [review]
